src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -1013,47 +1015,23 @@ private PackageLicenseInfo GeneratePackageLicenseInfo(IPackageSearchMetadata met
         /// <summary>
         /// Get the package metadata to see if RequireLicenseAcceptance is true
         /// </summary>
-        private async Task<List<IPackageSearchMetadata>> GetPackageMetadataAsync(
-            IEnumerable<SourceRepository> sources,
+        private async Task<List<IPackageSearchMetadata>> GetOnlyLocalPackageMetadataAsync(
+            IEnumerable<SourceRepository> localSources,
             IEnumerable<PackageIdentity> packages,
             CancellationToken token)
         {
             var results = new List<IPackageSearchMetadata>();
 
-            // local sources
-            var localSources = new List<SourceRepository>
-            {
-                _packageManager.PackagesFolderSourceRepository
-            };
-            localSources.AddRange(_packageManager.GlobalPackageFolderRepositories);
-
             var allPackages = packages.ToArray();
 
             using (var sourceCacheContext = new SourceCacheContext())
             {
-                // first check all the packages with local sources.
-#pragma warning disable CS8619 // Nullability of reference types in value doesn't match target type.
                 IPackageSearchMetadata[] completed = (await TaskCombinators.ThrottledAsync(
                     allPackages,
                     (p, t) => GetPackageMetadataAsync(localSources, sourceCacheContext, p, t),
-                    token)).Where(metadata => metadata != null).ToArray();
-#pragma warning restore CS8619 // Nullability of reference types in value doesn't match target type.
+                    token)).Cast<IPackageSearchMetadata>().Where(metadata => metadata != null).ToArray();
```

Comment:
Minor "reading understanding" improvement, though there's no functional difference.

```suggestion
                    token)).Where(metadata => metadata != null).Cast<IPackageSearchMetadata>().ToArray();
```

To me, casting to a non-nullable type tells me that there are already no nulls possible. If that happens before the `!= null` filter, that suggests to me that there can't be any nulls before that point, making the `!= null` check unnecessary (and in that case, `GetPackageMetadataAsync` has the wrong annotation).

By moving the cast to after the `!= null` filtering, it tells me that filtering the nulls is functionally important.